web/src/components/data-fields/address/AddressTextField.tsx

```diff
@@ -11,15 +12,25 @@ export interface Props extends Omit<GenericTextFieldProps, "value" | "error" | "
   label?: string;
   secondaryLabel?: string;
   errorBarType: "ALWAYS" | "MOBILE-ONLY" | "DESKTOP-ONLY" | "NEVER";
+  required?: boolean | false;
 }
 
 export const AddressTextField = ({ className, ...props }: Props): ReactElement => {
   const { state, setAddressData } = useContext(AddressContext);
   const { doesFieldHaveError } = useAddressErrors();
+  const { state: taxClearanceCertificateData, setTaxClearanceCertificateData } = useContext(
+    TaxClearanceCertificateDataContext
+  );
 
   const handleChange = (value: string): void => {
     props.handleChange && props.handleChange(value);
     setAddressData((prevAddressData) => ({ ...prevAddressData, [props.fieldName]: value }));
+    if (taxClearanceCertificateData !== undefined) {
```

Comment:
boulder - it's not clear to me why we are setting state in two different places. can you provide some context?

Reply:
This component is a shared component, the AddressContext is used to store the value. We should have fewer moving parts in this component; TaxClearanceCertificateDataContext should be removed all together.

Reply:
The address components are not using DataFormErrorMapContext. You should look into how alerts and error are firing in other components.